src/sync/sync_user.cpp

```diff
@@ -408,7 +410,7 @@ void SyncUser::set_binding_context_factory(SyncUserContextFactory factory)
 
 void SyncUser::refresh_custom_data(std::function<void(util::Optional<app::AppError>)> completion_block)
 {
-    SyncManager::shared().app()->refresh_custom_data(shared_from_this(), [completion_block](util::Optional<app::AppError> error){
+    m_sync_manager->app().lock()->refresh_custom_data(shared_from_this(), [completion_block](util::Optional<app::AppError> error){
```

Comment:
The only reason to use a weak pointer over a raw pointer is to be able to specifically handle the case where lock() returns null. If not checking it here is correct (and it appears to be), then SyncManager::m_app might as well be a raw pointer.

Reply:
Hmm... I suppose we should check. If the app has been deallocated, we should probably let the user know.